File: MiPrimeraAplicacion/app/src/main/java/com/micky/miprimeraaplicacion/backend/BaseDatos.java
package com.micky.miprimeraaplicacion.backend;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;
public class BaseDatos extends SQLiteOpenHelper {
    String SQL_CREATE="CREATE TABLE personas(id INTEGER PRIMARY KEY AUTOINCREMENT,paterno VARCHAR(40), materno VARCHAR(40), nombre VARCHAR(60));";
    String SQL_DROP="DROP TABLE IF EXISTS personas;";

    public BaseDatos(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(SQL_CREATE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(SQL_DROP);
        db.execSQL(SQL_CREATE);
    }

    // Método para agregar una persona a la base de datos
    public void agregarPersona(String paterno, String materno, String nombre) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put("paterno", paterno);
        values.put("materno", materno);
        values.put("nombre", nombre);
        db.insert("personas", null, values);
        db.close();
    }

    // Método para obtener todas las personas de la base de datos
    public List<String> obtenerTodasLasPersonas() {
        List<String> listaPersonas = new ArrayList<>();
        SQLiteDatabase db = this.getWritableDatabase();
        Cursor cursor = db.rawQuery("SELECT * FROM personas", null);
        if (cursor.moveToFirst()) {
            do {
                String persona = "ID: " + cursor.getInt(0) +
                        ", Paterno: " + cursor.getString(1) +
                        ", Materno: " + cursor.getString(2) +
                        ", Nombre: " + cursor.getString(3);
                listaPersonas.add(persona);
            } while (cursor.moveToNext());
        }
        cursor.close();
        db.close();
        return listaPersonas;
    }

    // Método para actualizar los datos de una persona en la base de datos
    public void actualizarPersona(int id, String paterno, String materno, String nombre) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put("paterno", paterno);
        values.put("materno", materno);
        values.put("nombre", nombre);
        db.update("personas", values, "id = ?", new String[]{String.valueOf(id)});
        db.close();
    }

    // Método para eliminar una persona de la base de datos
    public void eliminarPersona(int id) {
        SQLiteDatabase db = this.getWritableDatabase();
        db.delete("personas", "id = ?", new String[]{String.valueOf(id)});
        db.close();
    }
}

File: EXAMENFINAL_SIS104/app/src/main/java/com/example/examenfinal/backend/BaseDatos.kt
package com.example.examenfinal.backend

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class BaseDatos(context: Context): SQLiteOpenHelper(context, BaseDatos.NOMBRE_BASE_DATOS, null, BaseDatos.VERSION_BASE_DATOS){

    companion object{
        private val VERSION_BASE_DATOS = 1
        private val NOMBRE_BASE_DATOS = "db_final_sis104"
        private val NOMBRE_TABLA = "usuarios"
        private val ID = "id"
        private val Nombre = "nombre"
        private val Email = "email"
        private val Genero = "genero"
        private val Estado = "estado"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        val CREAR_TABLA =
            "CREATE TABLE $NOMBRE_TABLA ($ID INTEGER PRIMARY KEY AUTOINCREMENT,$Nombre String,$Email String,$Genero String,$Estado String);"
        db?.execSQL(CREAR_TABLA)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        val BORRAR_TABLA = "DROP TABLE IF EXISTS $NOMBRE_TABLA"
        db?.execSQL(BORRAR_TABLA)
        onCreate(db)
    }

    fun addResultado(d: usuario): Boolean {
        val db = this.writableDatabase
        val values = ContentValues().apply {
            put(ID, d.id)
            put(Nombre, d.nombre)
            put(Email, d.email)
            put(Genero, d.genero)
            put(Estado, d.estado)
        }
        val _success = db.insertWithOnConflict(NOMBRE_TABLA, null, values, SQLiteDatabase.CONFLICT_REPLACE)
        db.close()
        return (_success != -1L)
    }

    val d_result: List<usuario>
        @SuppressLint("Range")
        get() {
            val userList = ArrayList<usuario>()
            val db = this.readableDatabase
            val selectQuery = "SELECT * FROM $NOMBRE_TABLA"
            val cursor = db.rawQuery(selectQuery, null)
            if (cursor != null) {
                if (cursor.moveToFirst()) {
                    do {
                        val user = usuario().apply {
                            id = cursor.getInt(cursor.getColumnIndex(ID))
                            nombre = cursor.getString(cursor.getColumnIndex(Nombre))
                            email = cursor.getString(cursor.getColumnIndex(Email))
                            genero = cursor.getString(cursor.getColumnIndex(Genero))
                            estado = cursor.getString(cursor.getColumnIndex(Estado))
                        }
                        userList.add(user)
                    } while (cursor.moveToNext())
                }
                cursor.close()
            }
            return userList
        }

}